src/app/shared/state/entry-wizard.service.ts

```diff
@@ -0,0 +1,145 @@
+
+  /**
+   * Updates the list of git registries
+   */
+  updateGitRegistryStore() {
+    this.entryWizardStore.setLoading(true);
+    this.defaultService
+      .getUserRegistries()
+      .pipe(finalize(() => this.entryWizardStore.setLoading(false)))
+      .subscribe(
+        (registries: Array<string>) => {
+          this.entryWizardStore.update({ gitRegistries: registries });
+        },
+        (error: HttpErrorResponse) => {
+          this.alertService.detailedError(error);
+        }
+      );
+  }
+
+  /**
+   * Updates the list of organizations based on the given git registry
+   * @param registry git registry
+   */
+  updateGitOrganizationStore(registry: string) {
+    this.entryWizardStore.setLoading(true);
+    this.entryWizardStore.update({ gitOrganizations: undefined, gitRepositories: undefined });
+    const registryEnum = this.convertSourceControlStringToEnum(registry);
+    this.defaultService
+      .getUserOrganizations(registryEnum)
+      .pipe(finalize(() => this.entryWizardStore.setLoading(false)))
+      .subscribe(
+        (organizations: Array<string>) => {
+          this.entryWizardStore.update({ gitOrganizations: organizations });
+        },
+        (error: HttpErrorResponse) => {
+          this.alertService.detailedError(error);
+        }
+      );
+  }
+
+  /**
+   * Updates the list of repositories based on the given git registry and organization
+   * @param registry git registry
+   * @param organization git organization
+   */
+  updateGitRepositoryStore(registry: string, organization: string) {
+    this.entryWizardStore.setLoading(true);
+    const registryEnum = this.convertSourceControlStringToEnum(registry);
+
+    this.defaultService
+      .getUserOrganizationRepositories(registryEnum, organization)
+      .pipe(finalize(() => this.entryWizardStore.setLoading(false)))
+      .subscribe(
+        (repositories: Array<Repository>) => {
+          this.entryWizardStore.update({ gitRepositories: repositories });
+        },
+        (error: HttpErrorResponse) => {
+          this.alertService.detailedError(error);
+        }
+      );
+  }
+
+  /**
+   * Adds a workflow for the given git repository
+   * @param registry git registry
+   * @param organization git organization
+   * @param repositoryName git repository name
+   */
+  addWorkflowToDatabase(registry: string, organization: string, repositoryName: string) {
+    this.entryWizardStore.setLoading(true);
+    const registryEnum = this.convertSourceControlStringToEnum(registry);
+    this.defaultService
+      .addWorkflow(registryEnum, organization, repositoryName)
+      .pipe(finalize(() => this.entryWizardStore.setLoading(false)))
+      .subscribe(
+        (workflow: BioWorkflow) => {
+          this.alertService.detailedSuccess('Workflow ' + registry + '/' + organization + '/' + repositoryName + ' has been added');
+          this.updateGitRepositoryStore(registry, organization);
```

Comment:
Could you just update the store with this one item instead of fetching everything again?

Reply:
The store isn't a list of workflows but rather a list of repository objects.

Reply:
Oh realized this is in the success section, I think I can remove this and keep it only in the error section.

Reply:
Talked with Gary, have a possible solution.